test/Microsoft.ML.Tests/Transformers/CustomMappingTests.cs

```diff
@@ -62,17 +62,6 @@ public void TestCustomTransformer()
             var tempoEnv = new MLContext(1);
             var customEst = new CustomMappingEstimator<MyInput, MyOutput>(tempoEnv, MyLambda.MyAction, "MyLambda");
 
-            try
-            {
-                TestEstimatorCore(customEst, data);
-                Assert.True(false, "Cannot work without RegisterAssembly");
-            }
-            catch (InvalidOperationException ex)
-            {
-                if (!ex.IsMarked())
-                    throw;
-            }
-            ML.ComponentCatalog.RegisterAssembly(typeof(MyLambda).Assembly);
```

Comment:
I wonder, what happens (and what should happen) if the user actually registers the assembly as done here? Will it throw an exception, or will it work anyway with the fix on this PR?

Reply:
In fact, I am unsure... what is the expected behavior if the user doesn't register the assembly but still wants to use the CustomTransformer without saving/loading it?

Reply:
It will work. The assembly can be registered multiple times.

Reply:
Great.

If anything, I would recommend using the InlineData trick I mentioned in [another comment](https://github.com/dotnet/machinelearning/pull/4989#discussion_r401428872) to test both cases: when the user registers the assembly manually (even if it's not necessary) and when they don't register it.

Reply:
Done in the next iteration.

---
In reply to: [401794399](https://github.com/dotnet/machinelearning/pull/4989#discussion_r401794399) [](ancestors = 401794399)